lib/linked_list.rb

```diff
@@ -12,120 +12,255 @@ def initialize(value, next_node = nil)
 
 # Defines the singly linked list
 class LinkedList
-    def initialize
-      @head = nil # keep the head private. Not accessible outside this class
+  def initialize
+    @head = nil # keep the head private. Not accessible outside this class
+  end
+
+  # method to add a new node with the specific data value in the linked list
+  # insert the new node at the beginning of the linked list
+  def add_first(value)
+    @head = Node.new(value, @head)
+  end
+
+  # method to find if the linked list contains a node with specified value
+  # returns true if found, false otherwise
+  def search(value)
+    current = @head
+
+    until current.nil?
+      return true if current.data == value
+      current = current.next
     end
+    return false
+  end
 
-    # method to add a new node with the specific data value in the linked list
-    # insert the new node at the beginning of the linked list
-    def add_first(value)
-      raise NotImplementedError
+  # method to return the max value in the linked list
+  # returns the data value and not the node
+  def find_max
+    return nil if @head.nil?
+    max_val = 0

```

Comment:
What if the list is of negative numbers?

```suggestion
    max_val = @head.data
```